pkg/tools/log/levelchange.go

```diff
@@ -0,0 +1,33 @@
+package log
+
+import (
+	"context"
+	"github.com/sirupsen/logrus"
+	"os"
+	"os/signal"
+	"syscall"
+)
+
```

Comment:
FYI: code can be simplified and we also can exclude linux signal defendency.

```suggestion go
func SetLogLevelChangeOnSignal(ctx context.Context,  signals map[os.Signal]logrus.Level) {
 changeLogLevelSginal := make(chan os.Signal, len(signals))
  for s :=range signals {
         signal.Notify(changeLogLevelSginal,  sig)
  }
  go func() {
  
   for {
      select {
         case <-ctx.Done():
          signal.Stop(changeLogLevelSginal)
          close(changeLogLevelSginal)
         return
         case sig <- changeLogLevelSginal:
            logrus.SetLevel(signals[sig])
      }
   }
  }
}
```